cryptosuite/jsonwebkey2020.go

```diff
@@ -270,3 +275,23 @@ func NewJSONWebKeyVerifier(kid string, key crypto.PublicKeyJWK) (*JSONWebKeyVeri
 		JWTVerifier: *verifier,
 	}, nil
 }
+
+// PubKeyBytesToTypedKey converts a public key byte slice to a crypto.PublicKey based on a given key type, merging
+// both LD key types and JWK key types
+func PubKeyBytesToTypedKey(keyBytes []byte, kt LDKeyType) (gocrypto.PublicKey, error) {
+	var convertedKeyType crypto.KeyType
+	switch kt.String() {
+	case JsonWebKey2020.String():
+		// we cannot know this key type based on the bytes alone
+		return keyBytes, nil
+	case crypto.Ed25519.String(), did.Ed25519VerificationKey2018.String(), did.Ed25519VerificationKey2020.String():
+		convertedKeyType = crypto.Ed25519
+	case crypto.X25519.String(), did.X25519KeyAgreementKey2019.String(), did.X25519KeyAgreementKey2020.String():
+		convertedKeyType = crypto.X25519
+	case crypto.SECP256k1.String(), did.EcdsaSecp256k1VerificationKey2019.String():
+		convertedKeyType = crypto.SECP256k1
+	default:
+		return nil, fmt.Errorf("unsupported key type: %s", kt)
+	}
+	return crypto.BytesToPubKey(keyBytes, convertedKeyType)
```

Comment:
type {
  keyBytes []byte
  keyType kt.String() type
}

would a wrapper object give much value? I guess you already have the keytype in hand so maybe not

Reply:
hmm...yeah we have `LDKeyType` and this is going to `KeyType`, but the interface that this method is most often being used with is expecting a `gocrypto.PublicKey`